convert.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/env bash
+cd "$(dirname ${BASH_SOURCE[0]})"
+
+printf "%s\n" \
+    "-=- Luxor to OpenSMCE Game Converter -=-" \
+    "" \
+    "===========================================" \
+    "Step 0. Checking the requirements..." \
+    "==========================================="
+good=1
+
+python --version > /dev/null
+if [ $? -ne 0 ]; then
+    printf "! Python is not installed !\n"
+    good=0
+fi
+
+if [ ! -d "./data" ]; then
+    printf "! Folder "data" does not exist !\n"
+    good=0
+fi
+
+if [ ! -d "./english" ]; then
+    printf "! Folder \"english\" does not exist !\n"
+    good=0
+fi
+
+if [ ! -d "./assets" ]; then
+    printf "! Folder \"assets\" does not exist !\n"
+    good=0
+fi
+
+if [ ! -d "./Luxor_appendix" ]; then
+    printf "! Folder \"/Luxor_appendix\" does not exist !\n"
+    good=0
+fi
+
+if [ $good -eq 1 ]; then
+    printf "All good!\n"
+else
+    printf "%s\n" \
+        "===========================================" \
+        "The converter found some issues and will not convert the game properly." \
+        "They are listed above." \
+        "Before proceeding, fix them and run convert.bat again." \
+        "If you think some of these errors are false positive," \
+        "contact me on Discord!" \
+        "===========================================" \
+        "" \
+        "The converter will now close."
+    exit
+fi
+
+printf "%s\n" '' '' \
+     "If you haven't checked README file yet, just in case please do it before proceeding!" \
+     "The original files will be available in \"_BACKUP\" directory. If you want, you can delete it afterwards." \
+     "===========================================" \
+     "The converter will start working once you press any key."
+
+read -rsn1
+
+printf "%s\n" '' '' '' '' \
+    "===========================================" \
+    "Step 1. Preparing and backing up..." \
+    "==========================================="
+
+if [ -d "./data/data" ]; then
+    printf "The \"data\" folder is nested too deeply, fixing...\n"
+    cp -aflt ./data ./data/data/*
+    rm -rf ./data/data
+    printf "Done!"
+fi
+
+if [ -d "./english/english" ]; then
+    printf "The \"english\" folder is nested too deeply, fixing...\n"
+    cp -aflt ./english ./english/english/*
+    rm -rf ./english/english
+    printf "Done!"
+fi
+
+mkdir -p ./_BACKUP
+cp -aft ./_BACKUP ./data ./english ./assets ./Luxor_appendix
+
+printf "%s\n" '' '' '' '' \
+    "===========================================" \
+    "Step 2. Merging data/ with english/..." \
+    "==========================================="
+
+    cp -aflt ./data ./english/data/*
+    rm -rf ./english
+
+
+printf "%s\n" '' '' '' '' \
+    "===========================================" \
+    "Step 3. Converting..." \
+    "==========================================="
+# Linux is case sensitive -_-
+mv data/maps/PooloftheLotusBlossom data/maps/PoolOfTheLotusBlossom
+mv data/maps/ScrollofThoth data/maps/ScrollOfThoth
+
+#python -m pip install pillow
```

Comment:
Yeah, the automatic package installing is something that is indeed a bit sketchy and I did consider removing it or at least blatantly prompting the user to agree before actual installation starts to not compromise automation.

There was this hacker attack somewhere in 2022, where malicious code was injected into some more popular Python packages. The Pillow library was not compromised, but if it was, this could pose a real security threat for an innocent OpenSMCE user. It's best to leave this commented out for now, but I need to amend the readme file to state that the user must download this package manually, and that we don't provide any guarantee that it will be 100% safe.